core/src/main/scala/s3/s3Stream.scala

```diff
@@ -186,11 +186,14 @@ trait S3StreamBuilder {
    * @param chunkUploadConcurrency chunk upload concurrency. Order is guaranteed even with chunkUploadConcurrency > 1.
    */
   def uploadStreamAsMultipartFile(bucket: String, getKey: Long => String, nbChunkPerFile: Int,
-      chunkUploadConcurrency : Int): Flow[ByteString, CompleteMultipartUploadResult, Unit] = {
+      chunkUploadConcurrency : Int): Flow[ByteString, CompleteMultipartUploadResult, akka.NotUsed] = {
 
     Flow[ByteString]
       .via(FlowExt.zipWithIndex)
-      .splitWhen { case (_, i) => i != 0 && i % nbChunkPerFile == 0 }
+      .splitWhen { x =>
+        val i = x._2
+        i != 0 && i % nbChunkPerFile == 0
+      }
```

Comment:
Scala compilier was complaining here about not being able to infer types on this partial function. `splitWhen` accepts a function, not a partial function, at any rate. Only real solution is to have a 1 argument function (which is what `splitWhen` wants) and then pull out the 2nd argument (the `long`).